### PR TITLE
:hammer: drivers interfaces

### LINKED ISSUE #103: Refactor driver interface / behaviour ?
Maybe we should refactor the driver so drivers returns this interface everytime:
 - getEnhancer: returns enhancer
 - getReducer: returns { path, reducer }
 - getDriver: returns driver

And we use them like that:
```js
export default createStore(
  description,
  {
    drivers: [
      http(), 
      form(state => state.form),
      reduxLittleRouter(routes, state => state.router, routerImpl),
    ]
  }
)
```

And `k-ramel` use the returned interface to configurate the store where it needs to be, enhancer would be pushed the same order as driver declaration.

### PR DESCRIPTION
**Pull request by fabienjuif**
fixes #103 

## TODO
 - [x] Change how k-ramel (core) handle drivers with the new interface
 - [x] Tests

### CODE REVIEW

**Comment by fabienjuif** [comment]
Right now it works with `redux-little-router` 🎉 🎉 
And it allow me to do : (on k-mille)

```diff
diff --git a/ui/ui-expenses/src/store/index.js b/ui/ui-expenses/src/store/index.js
index e9f93e1..af30a1f 100644
--- a/ui/ui-expenses/src/store/index.js
+++ b/ui/ui-expenses/src/store/index.js
@@ -2,31 +2,20 @@ import { initializeCurrentLocation } from 'redux-little-router'
 import { createStore, keyValue, simpleObject } from 'k-ramel'
 import drivers, { router } from './drivers'
 
-export default () => {
-  const store = createStore(
-    {
-      router: router.getReducer(),
-      data: {
-        profile: simpleObject(),
-        fileId: simpleObject({ defaultData: '' }),
-        prices: simpleObject({ defaultData: [] }),
-        expenses: keyValue({ key: 'id' }),
-      },
-      ui: {
-        header: simpleObject({ defaultData: { title: 'envoi' } }),
-        list: simpleObject({ defaultData: [] }),
-      },
+export default createStore(
+  {
+    data: {
+      profile: simpleObject(),
+      fileId: simpleObject({ defaultData: '' }),
+      prices: simpleObject({ defaultData: [] }),
+      expenses: keyValue({ key: 'id' }),
     },
-    {
-      enhancer: router.getEnhancer(),
-      drivers,
+    ui: {
+      header: simpleObject({ defaultData: { title: 'envoi' } }),
+      list: simpleObject({ defaultData: [] }),
     },
-  )
-
-  const initialLocation = store.getState().router
-  if (initialLocation) {
-    store.dispatch(initializeCurrentLocation(initialLocation))
-  }
-
-  return store
-}
+  },
+  {
+    drivers,
+  },
+)
```

File: packages/k-ramel/src/createStore/createStore.js [hunk 1]
```diff
@@ -28,9 +28,32 @@ export default (definition, options = defaultOptions) => {
...
+  const definitionWithDrivers = { ...definition }
+
+  // use drivers
+  const driversEnhancers = []
+  const driversInits = []
+  Object.values(drivers)
+    .forEach((driver) => {
+      // bind reducer to store definition
+      if (driver.getReducer) {
+        const { reducer, path } = driver.getReducer() // eslint-disable-line no-unused-vars
```
Comment (fabienjuif) [inline packages/k-ramel/src/createStore/createStore.js]: If someone find a way to do this a better way (and without using lodash)

**Comment by fabienjuif** [comment]
This is not related. But I try to move my reducer (path) to `ui.router` and I got:
image

**Comment by fabienjuif** [comment]
I will not fix coverage

File: packages/k-ramel/src/createStore/createStore.js (hunk 1 above)
Comment (bpetetot) [reply packages/k-ramel/src/createStore/createStore.js]: I don't know if it's a better way, but you don't have to use `eval` or  `lodash` :
```js
const reducerPath = path.split('.')
const reducerName = reducerPath.pop()
const reducerLocation = reducerPath.reduce((acc, next) => acc[next], definitionWithDrivers)
reducerLocation[reducerName] = reducer
```

Comment (fabienjuif) [reply packages/k-ramel/src/createStore/createStore.js]: I have to benchmark that

Comment (fabienjuif) [reply packages/k-ramel/src/createStore/createStore.js]: https://github.com/alakarteio/k-ramel/issues/132

**Pull request merged**